enumerables.rb

```diff
@@ -0,0 +1,112 @@
+# Creating the enumerable module
+module Enumerables 
+    #my_each method
+    def my_each 
+        i = 0
+        while i < self.length
+            yield(self[i])
+            i+= 1
+        end
+    end
+      
+#my_each_with_index method
+def my_each_with_index
+    index = 0
+    i = 0
+    while i < self.length
+    yield(self[i], index)
+    index += 1
+    i+= 1
+    end
+end
+
+#my_select method
+def my_select
+arr = []
+self.my_each {|el| arr << el if yield(el)}
+arr 
+end
+
+#my_all? method
+def my_all? 
```

Comment:
This method does not return a value.

In fact, you may have to reconsider the logic flow of this method. It's based on whether all elements meet the requirements on the passed block.

Consider the following:
- have result appear on line 41 instead (so you'd actually get it to return value returned
- consider that if at least one does not meet the requirements of the passed block that you should break... no need to check rest because that automatically makes it **false**, so maybe you can set result to **true** initially and check for a **negative** situation where you set it to false. 💭

Reply:
TIp 💡 : Maybe an unless maybe useful in this situation 🤔 💭